staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go

```diff
@@ -304,3 +306,13 @@ func Convert_Slice_string_To_Slice_int32(in *[]string, out *[]int32, s conversio
 	}
 	return nil
 }
+
+// Convert_Slice_string_To_v1_DeletionPropagation allows converting a URL query parameter propagationPolicy
+func Convert_Slice_string_To_v1_DeletionPropagation(input *[]string, out *DeletionPropagation, s conversion.Scope) error {
+	if len(*input) > 0 {
```

Comment:
Don't you need to check if `*input` is `nil` first?

Reply:
We don't check if `*input` and `*output` are `nil` in conversion functions, e.g. https://github.com/kubernetes/kubernetes/blob/7e75a09db6dc13218b4f5e06493115ae96f83774/staging/src/k8s.io/apimachinery/pkg/runtime/conversion.go
https://github.com/kubernetes/kubernetes/blob/7e75a09db6dc13218b4f5e06493115ae96f83774/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1beta1/conversion.go
I will take another look to see where is it enforced.

Reply:
It's enforced by 
https://github.com/kubernetes/kubernetes/blob/bed45311315f63f7b0ddce756559920c0139273e/staging/src/k8s.io/apimachinery/pkg/conversion/converter.go#L497
through
https://github.com/kubernetes/kubernetes/blob/bed45311315f63f7b0ddce756559920c0139273e/staging/src/k8s.io/apimachinery/pkg/conversion/helper.go#L27-L39

Reply:
Ok thanks 👍